src/mds/Server.cc

```diff
@@ -3517,11 +3517,15 @@ CInode* Server::prepare_new_inode(MDRequestRef& mdr, CDir *dir, inodeno_t useino
     }
   } else {
     _inode->gid = mdr->client_request->get_owner_gid();
-    ceph_assert(_inode->gid != (unsigned)-1);
   }
 
   _inode->uid = mdr->client_request->get_owner_uid();
-  ceph_assert(_inode->uid != (unsigned)-1);
+
+  if (_inode->gid == (unsigned)-1 || _inode->uid == (unsigned)-1) {
+    dout(0) << "WARNING: client specified uid " << _inode->uid << " gid " << _inode->gid << " for ino " << _inode->ino << dendl;
+    mds->clog->error() << mdr->client_request->get_source()
+       << " specified uid " << _inode->uid << " gid " << _inode->gid << " for ino " << _inode->ino;
```

Comment:
This could generate incredible amount of spam. Is this really an error or can we replace it with a reasonable default?

Reply:
This is an error and if we have `uid`/`gid` equal to `-1` it means that something is terribly wrong with the client. Ideally, we need to fail the request somehow. Xiubo (@lxbsz) what do you think about this?